### ISSUE TITLE
Error: vips_colourspace: no known route between 'multiband' and 'srgb' on rotate

### ISSUE THREAD

**Issue by automata** [lovell/sharp#375]
While trying to rotate [this particular 1x1 GIF file](http://i.imgur.com/5hmA0Dc.gif) on [noflo-sharp](https://github.com/noflo/noflo-sharp), we're getting the following error:

```
Error: vips_colourspace: no known route between 'multiband' and 'srgb'
```

The whole build is available at [this Travis build log](https://travis-ci.org/noflo/noflo-sharp/jobs/114901200#L571).

**Comment by automata** [comment]
Probably related with https://github.com/jcupitt/libvips/issues/172?

**Comment by lovell** [comment]
Hello, the following:
```javascript
sharp('5hmA0Dc.gif').rotate(90).png().toBuffer(console.log);
```
seems to work for me:
```
null <Buffer 89 50 4e ... > { format: 'png', width: 1, height: 1, channels: 4, size: 91 }
```
Are you able to provide a small code sample to help narrow this down?

**Comment by lovell** [comment]
@automata Are you able to share a code sample that fails?

**Comment by lovell** [comment]
@automata Is this still a problem?

**Comment by automata** [comment]
@lovell sorry for the long delay. I tested exactly the same code snippet you suggested, locally on Ubuntu 14.04, using latest sharp and I got:

```
[Error: vips_colourspace: no known route between 'multiband' and 'srgb'] null null
```
Is there any other information I could provide?

**Comment by lovell** [comment]
Thanks for the update. What is the output of `console.log(sharp.versions)`?

**Comment by automata** [comment]
```
{ vips: '8.2.3' }
```

**Comment by lovell** [comment]
Thanks Vilson. It looks like you're using a globally-installed version of libvips compiled against a version of *magick that doesn't play nicely with this file. What does `convert -version` say, so I can try to reproduce this using the same *magick version/fork as you?

**Comment by automata** [comment]
```
Version: ImageMagick 6.7.7-10 2014-03-06 Q16 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2012 ImageMagick Studio LLC
Features: OpenMP
```

**Comment by paras20xx** [comment]
Just to add weight to this issue, we are also facing this.
I'll check the versions on my system and update here.

Link to the problematic image (2x1 size): http://i.imgur.com/gZ5jlmE.gif

**Comment by papandreou** [comment]
@lovell, it's reproducible with @automata's image on Travis (Ubuntu 12.04 LTS Server Edition 64 bit according to [their documentation](https://docs.travis-ci.com/user/ci-environment/)) with nothing but sharp 0.14.1 installed: https://travis-ci.org/papandreou/express-processimage/builds/125588397

**Comment by lovell** [comment]
Thanks for the extra reports confirming the problem.

As a related aside, libvips v8.3.1 will support giflib v5, so I plan for v0.15.0 of sharp to bundle it and no longer bundle *magick. You'll still be able to use the *magick loaders with sharp+libvips shared library compiled with *magick support, but I consider the attack surface and support burden of the *magick forks too large.

**Comment by lovell** [comment]
I've reproduced this with the `gZ5jlmE.gif` image using sharp v0.14.1 and the bundled GraphicsMagick v1.3.23.

The result of `sharp('gZ5jlmE.gif').metadata()` reports the following, which [suggests](https://github.com/jcupitt/libvips/blob/master/libvips/foreign/magick2vips.c#L363) that GraphicsMagick thinks it's a 2 channel RGB image :(

```
{ format: 'magick',
  width: 2,
  height: 1,
  space: 'rgb',
  channels: 2,
  hasProfile: false,
  hasAlpha: false }
```
I suspect this might be a GraphicsMagick/MagickWand bug and needs more investigation. I plan to work on integration with giflib v5 as part of #369 in preference to looking at this further. I'm happy to chip in with comments should anyone wish to report this upstream.

(I also spotted that `sharp.versions` is broken in v0.14.1; should probably add a test for that.)
